driver/src/driver/stablex_driver.rs

```diff
@@ -39,14 +39,21 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
-            info!("Successfully applied solution to batch {}", batch);
-            true
+        let current_objective_value = self.contract.get_current_objective_value()?;
+        let submitted = if let Some(surplus) = solution.surplus {
```

Comment:
I was searching here for a more elegant solution, but I can not find it.

Thread discussing it:
https://github.com/rust-lang/rfcs/issues/929#issuecomment-81355948

Reply:
You are comparing surplus (should probably be renamed to utility in a separate PR) with the current_objective_value (which at the moment is volume). I don't think this works in the current setting as a solution with little surplus would not be corrected as long as it had more volume than a new solution had utility. Can't we compute and compare the proper volume instead?

As for how to make this more elegant with less nested if statements, you could have a helper method that returns a `SolutionSubmission` enum (e.g. Trivial, NotGoodEnough, Submit) and then have a match statement with only one level.

Reply:
I thought, although this is not the correct solution, it is better than checking just that is is not the trivial solution.
Not sure, whether it is worth to code the volume formula ( actually the fee formula), as the rounding that would be needed to be considered is non-trivial.

But yeah, I am happy to pause this PR and keep it for later, once Tom's solver and the smart contract calculate the same objective value.

Reply:
You could use match statements with conditional patterns, something like this:
```rust
        let submitted = match solution.surplus {
            Some(surplus) if current_objective_value < surplus => {
                self.contract.submit_solution(batch, orders, solution)?;
                info!("Successfully applied solution to batch {}", batch);
                true
            },
            _ => {
                info!("Not submitting trivial solution for batch {}", batch);
                false
            },
        };
```

Reply:
You could also use `Option::map` and do something like:
```rust
        let submitted = if Some(true) = solution.surplus.map(|surplus| current_objective_value < surplus) {
            self.contract.submit_solution(batch, orders, solution)?;
            info!("Successfully applied solution to batch {}", batch);
            true
        } else {
            info!("Not submitting trivial solution for batch {}", batch);
            false
        };
```